tests/cli/test_datasets.py

```diff
@@ -1694,6 +1694,50 @@ def test_lfs_hook_can_be_avoided(runner, project, subdirectory, large_file):
     assert "OK" in result.output
 
 
+def test_datadir_hook(runner, client, subdirectory):
+    """Test pre-commit hook fir checking datadir files."""
+    datadir = client.path / "test"
+    datadir.mkdir()
+
+    result = runner.invoke(cli, ["--no-external-storage", "dataset", "create", "--datadir", str(datadir), "my-dataset"])
+    assert 0 == result.exit_code, format_result_exception(result)
+
+    file = datadir / "new_file"
+    file.write_text("some updates")
+    file2 = datadir / "another_file"
+    file2.write_text("some updates")
+
+    client.repository.add(all=True)
+
+    # Commit fails when file is not tracked in LFS
```

Comment:
If the file was committed but not tracked in LFS then things would have passed, right?

Reply:
ah that's a copy paste error, the comment was from a similar piece of code in another test :see_no_evil: